### ISSUE TITLE
Add afterEachFailedTest callback

### ISSUE THREAD

**Issue by xgouchet** [JetBrains/spek#216]
The `afterEachTest` callback is great to cleanup resources, but it lacks information on the success or failure of the test after which it's invoked. 

To have a concrete example, I want to add logs in the standard output stream in the afterEachTest callback when the test fails. 

Right now I can add my logs, but it will apply  to all the test, whether or not they fail, and this will clutter my output stream with needless information. 

I'm not sure how this could be possible, maybe adding an `afterEachFailedTest` callback ?

**Comment by raniejade** [comment]
I'm not really keen on adding new methods, we could just add a parameter to the `afterEachTest` blocks that contains information about the test (description, failure, etc..).

**Comment by xgouchet** [comment]
@raniejade yes adding context information as a parameter would totally work also.
